subsys/net/lib/openthread/platform/settings.c

```diff
@@ -252,10 +252,7 @@ otError otPlatSettingsSet(otInstance *aInstance, uint16_t aKey,
 
 	LOG_DBG("%s Entry aKey %u", __func__, aKey);
 
-	(void)ot_setting_delete_subtree(aKey, -1);
```

Comment:
I think we still need to remove the old entries, children of `OT_SETTINGS_ROOT_KEY` / `aKey`. Otherwise, if this code is run against settings created using the old implementation, it may happen that a subsequent `otPlatSettingsGet` will stil return the old value.